### PR TITLE
[CARBONDATA-1756] Improve Boolean data compress rate by changing RLE to SNAPPY algorithm

### PR DESCRIPTION
**Pull request by xubo245**
Improve Boolean data compress rate by changing RLE to SNAPPY algorithm
Because Boolean data compress rate that uses RLE algorithm is lower than SNAPPY algorithm in most scenario.

This PR only changed about 4 lines for compress algorithm, and we also add some test cases for testing  Boolean data compress rate.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
    TestBooleanCompressSuite.scala
    For 100000000 booean data and true value is 50%, random, the compress rate of SNAPPY algorithm  is  13.59735( 549995608 Byte before compressing/40448723 Byte of after compressing), But RLE is 4.115418 (549993780/133642270)

 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1223/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1237/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1251/

**Comment by xubo245** [comment]
@jackylk  Please review it

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1403/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1848/

File: integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/compress/BooleanFile.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.spark.testsuite.booleantype.compress
+
+import java.io.{File, PrintWriter}
+
+import scala.util.Random
+
+object BooleanFile {
```
Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/compress/BooleanFile.scala]: If it is only used in TestBooleanCompressSuite, you can make it private function inside TestBooleanCompressSuite

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1464/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1881/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1467/

File: integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/compress/BooleanFile.scala (hunk 1 above)
Comment (xubo245) [reply integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/compress/BooleanFile.scala]: Yes, I have moved it into TestBooleanCompressSuite. Please review it again.

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**